pwndbg/heap/structs.py

```diff
@@ -0,0 +1,593 @@
+import ctypes
+import gdb
+
+import pwndbg.arch
+import pwndbg.heap
+import pwndbg.typeinfo
+import pwndbg.memory
+import pwndbg.glibc
+
+NBINS = 128
+BINMAPSIZE = 4
+TCACHE_MAX_BINS = 64
+
+if pwndbg.arch.ptrsize == 4:
```

Comment:
This is problematic, we need to invoke this when arch changes in GDB

(One can do sth like `attach <pid>` to attach to different architecture in GDB)

Reply:
I try to fix it in https://github.com/pwndbg/pwndbg/pull/1029/commits/f6cbc3fc548ed4f603f05b510307b097f43b3163
I re-initialize the heap when `pwndbg.event.stop`, and I add a decorator to reload `pwndbg.heap.structs` every time we create a new instance of `HeuristicHeap`.
Does this fix the problem?